### PR TITLE
New ci-setup command

### PR DESCRIPTION
**Pull request by talarian1**
- [x] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I suggest we add the following text before the first command prompt. Let's color the two titles.

**About this command**
This command sets up a basic CI pipeline which uses the JFrog Platform.
It currently supports maven, gradle and npm, but additional package managers will be added in the future.
The generated CI pipeline is based on JFrog Pipelines, but additional CI providers will be added in the future.

**Important**
1. When asked to provide credentials for your JFrog Platform and Git provider, please make sure the credentials have admin privileges.
2. You can exit the command by hitting 'control + C' at any time. The values you provided before exiting are saved (with the exception of passwords and tokens) and will be set as defaults the next tine you run the command.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
1. If yuo provide a branch that does not exist, no error message is prompted.
2. [Error] ./jfrog-vcs-workspace isn't empty. --> [Error] The ./jfrog-vcs-workspace directory isn't empty.

File: general/cisetup/pipelinesyaml.go [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+	"fmt"
+	"github.com/jfrog/jfrog-cli-core/utils/coreutils"
+	"github.com/jfrog/jfrog-cli/utils/cliutils"
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"gopkg.in/yaml.v2"
+	"strconv"
+	"strings"
+)
+
```
Comment (eyalbe4) [inline general/cisetup/pipelinesyaml.go:13]: I suggest we orginize the code in this file inside a struct named JFrogPipelinesYamlGenerator. This struct can have one public entry point named Generate.

Let's do the same for the code inside ```pipelinesrun```. We can perhaps name the struct there ```JFrogPipelinesConfigurator```.

Both structs should move to jfrog-cli-core IMO.

File: general/cisetup/utils.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+func GetAllRepos(serviceDetails *utilsconfig.ServerDetails, repoType, packageType string) (*[]services.RepositoryDetails, error) {
+	servicesManager, err := artUtils.CreateServiceManager(serviceDetails, false)
+	if err != nil {
+		return nil, err
+	}
+	filterParams := services.RepositoriesFilterParams{RepoType: repoType, PackageType: packageType}
+	return servicesManager.GetAllRepositoriesFiltered(filterParams)
+}
+func GetVirtualRepo(serviceDetails *utilsconfig.ServerDetails, repoKey string) (*services.VirtualRepositoryBaseParams, error) {
```
Comment (eyalbe4) [inline general/cisetup/utils.go:37]: Missing new-line before function.

File: utils/cliutils/commandsflags.go [hunk 3]
```diff
@@ -1122,14 +1124,18 @@ var flagsMap = map[string]cli.Flag{
...
-	usersGroups: cli.StringFlag{
-		Name:  usersGroups,
+	UsersGroups: cli.StringFlag{
+		Name:  UsersGroups,
 		Usage: "[Optional] A list of comma-separated groups for the new users to be associated with.` `",
 	},
-	replace: cli.BoolFlag{
-		Name:  replace,
+	Replace: cli.BoolFlag{
+		Name:  Replace,
```
Comment (eyalbe4) [inline utils/cliutils/commandsflags.go:1132]: Flag should have [Default: false] and not [Optional].

File: utils/cliutils/commandsflags.go [hunk 4]
```diff
@@ -1122,14 +1124,18 @@ var flagsMap = map[string]cli.Flag{
...
 		Usage: "[Optional] Set to true if you'd like existing users or groups to be replaced.` `",
 	},
+	Admin: cli.BoolFlag{
+		Name:  Admin,
+		Usage: "[Optional] Set to true if you'd like to create an admin user.` `",
```
Comment (eyalbe4) [inline utils/cliutils/commandsflags.go:1137]: Flag should have [Default: false] and not [Optional].

File: general/cisetup/cisetup.go [hunk 5]
```diff
@@ -0,0 +1,753 @@
...
+	ConfigServerId           = "vcs-integration-platform"
+	VcsConfigFile            = "jfrog-cli-vcs.conf"
+	DefaultFirstBuildNumber  = "0"
+	DefaultWorkspace         = "./jfrog-vcs-workspace"
+	pipelineUiPath           = "ui/pipelines/myPipelines/default/"
+	permissionTargetName     = "ide-pt"
+	permissionTargetTemplate = `{"build":{"include-patterns":"**","actions-groups":{"%s":"read"}},"name":"%s"}`
+	pttFileName              = "ci-setup-ptt"
+	ideGroupName             = "ide-group"
+	ideUserName              = "ide-user"
```
[the remark this answers is not in the document]
Comment (RobiNino) [reply general/cisetup/cisetup.go:52]: I think we better come up with a name that is a bit more unique so the chances of such user existing prior to the setup are low

[the remark this answers is not in the document]
Comment (eyalbe4) [reply general/cisetup/cisetup.go]: OK - let's change it to "ide-developer" then.

**Pull request merged**